src/components/SavedRecipes.tsx

```diff
@@ -21,6 +26,34 @@ const SavedRecipes = () => {
       });
   }, []);
 
+  const handleRecipeSearch = useCallback((params: string) => {
+    const searchedRecipes = recipes.filter((recipe: SavedRecipe) => {
+        const search = params.toLowerCase();
+        const nameSearch = recipe.recipeName.toLowerCase();
+        const ingredientSearch = recipe.recipeIngredients.map((ingredient: { ingredientName: string }) => {
+            return ingredient.ingredientName.toLowerCase();
+        })
+        const tagSearch = recipe.recipeTags.map((tag: { tagName: string; }) => {
+            return tag.tagName.toLowerCase();
+        })
+        return nameSearch.includes(search) || ingredientSearch.includes(search) || tagSearch.includes(search);
+    })
+    if (searchedRecipes.length > 0) {
+      setFilteredRecipes(searchedRecipes);
+      setShow(true);
+    } else if (!searchedRecipes.length) {
+      setShow(false);
+    }
+  }, [recipes]);
+
+  useEffect(()=> {
+  if(params) {
+    handleRecipeSearch(params);
+  } else {
+    setSearchParams({search: ""});
+  }
+}, [searchParams, recipes]);
```

Comment:
NON-BLOCKING: I received this warning in the terminal 
```
src/components/SavedRecipes.tsx
  Line 55:4:  React Hook useEffect has missing dependencies: 'handleRecipeSearch', 'params', and 'setSearchParams'. Either include them or remove the dependency array  react-hooks/exhaustive-deps
```

Reply:
Thank you! I will add!